### ISSUE TITLE
Takes time to sync after a long time without transactions

### ISSUE THREAD

**Issue by ejpcmac** [voisine/breadwallet#300]
After a long time without doing transactions, breadwallet takes time to sync. This makes sense. But when the wallet is sync’ed, if you close the app and open it again without doing a transaction, it restarts to sync from the last transaction (and it takes time again).

Would it be possible to save the last scanned block to avoid this behaviour?

**Comment by dabura667** [comment]
> Would it be possible to save the last scanned block to avoid this behaviour?

That's how it works. Or at least suppose to.

I have never had the problems you describe. Could you explain in more detail?

**Comment by ejpcmac** [comment]
I didn’t had made transactions from or to my breadwallet for a few months until yesterday. During this period, if I opened the app, waited it sync, closed the app and re-opened it, it started always to sync from the same block, as if I didn’t had opened the app. Since I made a new transaction, the sync is quick. But I can’t say if it starts from the last transaction block or from the last time I opened the app.

**Comment by dabura667** [comment]
> it started always to sync from the same block, as if I didn’t had opened the app.

So you are looking at the log in xcode? Can you paste the log output here?

**Comment by ejpcmac** [comment]
I saw this in the app itself by touching the breadwallet logo (you can see a bubble showing the evolution of the synchronisation, and it always started from a block in the 350000s, even after having scanned to the 380000s). Unfortunately I have not checked the log, I’m not (yet) an iOS developer. Sorry for my lack of precision in the description of this issue.

**Comment by voisine** [comment]
@ejpcmac I did actually notice this myself when testing a colleagues wallet. It can happen if core data takes longer than 20-30seconds to persist to disk, in which case iOS will kill the app. I've improved core data performance, and also it now saves a single checkpoint block (which will persist to disk quickly) during syncing each time it passes a difficulty transition. Since you're not an iOS developer, that means it's fixed in the code and you won't have this problem after the next app update.

fixed by 8101ca7
